src/Compilers/Core/Portable/Emit/EditAndContinue/DeltaMetadataWriter.cs

```diff
@@ -539,60 +539,75 @@ protected override void CreateIndicesForNonTypeMembers(ITypeDefinition typeDef)
 
             int typeRowId = _typeDefs.GetRowId(typeDef);
 
+            // First we find the deleted methods, and add them to our dictionary. This is used later when
+            // processing events, properties, methods, and references.
+            var deletedMethods = _changes.GetDeletedMethods(typeDef);
+            if (deletedMethods.Length > 0)
+            {
+                var deletedTypeMembers = ImmutableDictionary.CreateBuilder<IMethodDefinition, DeletedMethodDefinition>(ReferenceEqualityComparer.Instance);
+                foreach (var methodDef in deletedMethods)
+                {
+                    var oldMethodDef = (IMethodDefinition)methodDef.GetCciAdapter();
+                    deletedTypeMembers.Add(oldMethodDef, new DeletedMethodDefinition(oldMethodDef, typeDef, _typesUsedByDeletedMembers));
+                }
+
+                _deletedTypeMembers.Add(typeDef, deletedTypeMembers.ToImmutableDictionary());
+            }
+
             foreach (var eventDef in typeDef.GetEvents(this.Context))
             {
                 if (!_eventMap.Contains(typeRowId))
                 {
                     _eventMap.Add(typeRowId);
                 }
 
-                this.AddDefIfNecessary(_eventDefs, eventDef);
+                var eventChange = _changes.GetChange(eventDef);
+                eventChange = _changes.FixSymbolChangeForReAddedMembers(eventDef, eventChange, DefinitionExistsInAnyPreviousGeneration);
```

Comment:
>FixSymbolChangeForReAddedMembers [](http://example.com/codeflow?start=39&length=32)

Seems like each call to FixSymbolChangeForReAddedMembers is preceded by _changes.GetChange, so we can move GetChange to FixSymbolChangeForReAddedMembers  and rename it to GetSymbolChangeForReAddedMembers

Reply:
Yeah, that should work. I originally tried to put all the logic in `GetChange` itself, so everything just worked magically, but that broke too many other things